### PR TITLE
SBOM dependency mapping NuGet

### PR DESCRIPTION
**Pull request by karthika-g**
Also has bug fix for finding artifactory repo info for nuget

### CODE REVIEW

File: src/LCT.PackageIdentifier/NugetProcessor.cs [hunk 1]
```diff
@@ -495,9 +449,33 @@ private static void ConvertToCycloneDXModel(List<Component> listComponentForBOM,
...
+                {
+                    GetDependencyDetails(components, prop, ref dependencies);
+                }
             }
         }
 
+        private static void GetDependencyDetails(Component compnent, NugetPackage prop,ref List<Dependency> dependencies)
+        {
+            List<Dependency> subDependencies = new();
+            foreach (var item in prop?.Dependencies)
```
Comment (sumanthkb44) [inline src/LCT.PackageIdentifier/NugetProcessor.cs:462]: i don't think Null check is required here ,Bcz while calling only u checked prop.Dependencies !=null

File: src/LCT.PackageIdentifier/NugetProcessor.cs [hunk 2]
```diff
@@ -656,19 +636,31 @@ private static List<NugetPackage> ConvertContainerAsNugetPackage(List<Container>
             {
                 foreach (var lst in containermodule.Components)
                 {
+                    List<string> depvalue =new List<string>();
+                    var dep = lst.Value.Dependencies.Count > 0 ? lst.Value.Dependencies : null;
```
Comment (sumanthkb44) [inline src/LCT.PackageIdentifier/NugetProcessor.cs:640]: check for lst.Value?.Dependencies

File: src/LCT.PackageIdentifier.UTest/NugetParserTests.cs [hunk 3]
```diff
@@ -50,29 +50,6 @@ public void ParsePackageConfig_GivenAInputFilePath_ReturnsSuccess()
...
-
-            //Act
-            List<NugetPackage> listofcomponents = NugetProcessor.ParsePackageLock(packagefilepath, appSettings);
-
-            //Assert
-            Assert.That(expectednoofcomponents, Is.EqualTo(listofcomponents.Count), "Checks for no of components");
-
-        }
         [TestCase]
         public void InputFileIdentifaction_GivenARootPath_ReturnsSuccess()
```
Comment (sumanthkb44) [inline src/LCT.PackageIdentifier.UTest/NugetParserTests.cs:54]: look for possible UT case for Nuget Dependncy checking

File: src/LCT.PackageIdentifier/NugetProcessor.cs (hunk 2 above)
Comment (karthika-g) [reply src/LCT.PackageIdentifier/NugetProcessor.cs:640]: un wanted declaration.Removed

File: src/LCT.Common/appSettings.json [hunk 4]
```diff
@@ -20,10 +21,10 @@
   "PackageFilePath": "/PathToInputDirectory", //For Docker run set as /mnt/Input
   "BomFolderPath": "/PathToOutputDirectory", //For Docker run set as /mnt/Output
   "BomFilePath": "/PathToOutputDirectory/<SW360 Project Name>_Bom.cdx.json",
-  //IdentifierBomFilePath : For multiple project type
+  //IdentifierBomFilePath : For multiple project type 
   "IdentifierBomFilePath": "",
   //CycloneDxBomFilePath: For Customer maintained SBOM
```
[the remark this answers is not in the document]
Comment (karthika-g) [reply src/LCT.Common/appSettings.json:26]: ok

File: src/LCT.PackageIdentifier.UTest/NugetParserTests.cs (hunk 3 above)
Comment (karthika-g) [reply src/LCT.PackageIdentifier.UTest/NugetParserTests.cs:54]: added

File: src/LCT.PackageIdentifier/NugetProcessor.cs (hunk 1 above)
Comment (karthika-g) [reply src/LCT.PackageIdentifier/NugetProcessor.cs]: removed

**Pull request merged**